src/mpi/init/initthread.c

```diff
@@ -358,6 +358,9 @@ int MPIR_Init_thread(int *argc, char ***argv, int required, int *provided)
     int thread_provided = 0;
     int exit_init_cs_on_failure = 0;
     MPIR_Info *info_ptr;
+
+    MPIR_T_env_init();
```

Comment:
Hmm... I guess I am adding them here and removing them in another commits. By the way, github insists ordering the commits by chronological order (creation time) which do not really reflect the intended order as I heavily re-ordered them in this PR. Maybe it is easier for me to start a new PR and reconstruct the commits, since now I had much clearer idea than I first started it. I'll discuss with you tomorrow.